src/integrations/util/makeServiceContextFromDependencies.ts

```diff
@@ -53,10 +53,14 @@ export default async function makeServiceContextFromDependencies(
 
   return resolveObj(
     Object.fromEntries(
-      dependencies.map((dependency) => [
-        `@${dependency.outputKey}`,
-        dependencyContext(dependency),
-      ])
+      dependencies.map((dependency) => {
+        const context =
+          dependency.isOptional && !dependency.configId
+            ? null
+            : dependencyContext(dependency);
```

Comment:
NIT: I think it's more readable to have `null` be the falsy path.

```suggestion
        const context =
          dependency.configId && !dependency.isOptional
            ? dependencyContext(dependency)
            : null;
```